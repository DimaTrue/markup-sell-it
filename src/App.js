import React from 'react';
import './styles/App.scss';
import Login from './components/LoginPage/Login/Login';
import WrongPath from './components/WrongPath/WrongPath';
import ProductList from './containers/productList';
import ProductItemPage from './containers/productItemPage';
import { BrowserRouter, Switch, Route } from "react-router-dom";
import ErrorBoundary from './components/CommonComponents/ErrorBoundary/ErrorBoundary';


class App extends React.Component {
  state = {
    isLoading: false,
    isAuthentificated: false,
  }

  login = () => {
    this.setState({ isAuthentificated: true });
  };

  logout = () => {
    this.setState({ isAuthentificated: false });
  };

  render() {
    return (
      <BrowserRouter>
        <Switch>
          <Route path='/login' component={() => <ErrorBoundary><Login /></ErrorBoundary>} />
          <Route path='/' exact component={() => <ErrorBoundary><ProductList /> </ErrorBoundary>} />
          <Route path='/product/:id' component={ProductItemPage} />
          <Route component={() => <ErrorBoundary><WrongPath /></ErrorBoundary>} />
        </Switch>
      </BrowserRouter>
    )
  }
}

export default App;