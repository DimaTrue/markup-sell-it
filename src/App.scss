@import './variables.scss';

/*....................base................*/
* {
  box-sizing: border-box;
  border: none;
}

body {
  margin: 0;
  padding: 0;
  background-color: $base-bg;
  font-family: Arial;
  font-weight: bold;
  font-size: 18px;
  color: $font-color;

  a {
    text-decoration: none;
  }
}

.container {
  max-width: 1440px;
  margin: 0 auto 0;
}

.form {
  @extend %form-style;

  &__btn {
    @extend %all-buttons;

    &:hover {
      border-bottom: 3px solid $border-color;
    }

  }
}