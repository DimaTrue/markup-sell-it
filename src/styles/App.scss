@import './variables.scss';

/*....................base................*/
* {
  box-sizing: border-box;
  border: none;
}

body {
  margin: 0;
  padding: 0;
  background-color: $base-bg;
  font-family: Arial;
  font-weight: bold;
  font-size: 18px;
  color: $font-color;

  a {
    text-decoration: none;
  }
}

