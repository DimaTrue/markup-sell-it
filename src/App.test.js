import React from 'react';
import ReactDOM from 'react-dom';
import { Router } from "react-router-dom";
import App from './App';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


describe ('renders', () => {

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render Router ', () => {
  const component = shallow(<App />);
  expect(component.find(Router)).toHaveLength(1);
}); 

it('match snapshot', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
})

});
