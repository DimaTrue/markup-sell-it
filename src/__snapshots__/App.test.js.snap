// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <UNDEFINED
        fallback={<Loading />}
      >
        <Switch>
          <Route
            component={[Function]}
            path="/login"
          />
          <UNDEFINED
            component={[Function]}
            exact={true}
            path="/"
          />
          <UNDEFINED
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProductItem)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/product/:id"
          />
          <UNDEFINED
            component={[Function]}
            path="/add_product"
          />
          <UNDEFINED
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(LogProps)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/profile"
          />
          <Route
            component={[Function]}
            path="/test"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </UNDEFINED>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            path="/login"
          />
          <UNDEFINED
            component={[Function]}
            exact={true}
            path="/"
          />
          <UNDEFINED
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProductItem)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/product/:id"
          />
          <UNDEFINED
            component={[Function]}
            path="/add_product"
          />
          <UNDEFINED
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(LogProps)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            path="/profile"
          />
          <Route
            component={[Function]}
            path="/test"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        "fallback": <Loading />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/login"
            />,
            <UNDEFINED
              component={[Function]}
              exact={true}
              path="/"
            />,
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProductItem)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/product/:id"
            />,
            <UNDEFINED
              component={[Function]}
              path="/add_product"
            />,
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(LogProps)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/profile"
            />,
            <Route
              component={[Function]}
              path="/test"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProtectedRoute)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "_status": 1,
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProductItem)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "path": "/product/:id",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProtectedRoute)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "_status": 1,
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "path": "/add_product",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProtectedRoute)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "_status": 1,
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(LogProps)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "path": "/profile",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(ProtectedRoute)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "_status": 1,
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.suspense),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <UNDEFINED
          fallback={<Loading />}
        >
          <Switch>
            <Route
              component={[Function]}
              path="/login"
            />
            <UNDEFINED
              component={[Function]}
              exact={true}
              path="/"
            />
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProductItem)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/product/:id"
            />
            <UNDEFINED
              component={[Function]}
              path="/add_product"
            />
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(LogProps)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/profile"
            />
            <Route
              component={[Function]}
              path="/test"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </UNDEFINED>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              path="/login"
            />
            <UNDEFINED
              component={[Function]}
              exact={true}
              path="/"
            />
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProductItem)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/product/:id"
            />
            <UNDEFINED
              component={[Function]}
              path="/add_product"
            />
            <UNDEFINED
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(LogProps)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/profile"
            />
            <Route
              component={[Function]}
              path="/test"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
          "fallback": <Loading />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/login"
              />,
              <UNDEFINED
                component={[Function]}
                exact={true}
                path="/"
              />,
              <UNDEFINED
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(ProductItem)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/product/:id"
              />,
              <UNDEFINED
                component={[Function]}
                path="/add_product"
              />,
              <UNDEFINED
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(LogProps)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/profile"
              />,
              <Route
                component={[Function]}
                path="/test"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProtectedRoute)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "_status": 1,
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProductItem)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "path": "/product/:id",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProtectedRoute)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "_status": 1,
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "path": "/add_product",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProtectedRoute)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "_status": 1,
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(LogProps)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "path": "/profile",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(ProtectedRoute)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "_status": 1,
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": Symbol(react.suspense),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
